tests/topologyDescription/topologyDescription-getServers-001.phpt

```diff
@@ -0,0 +1,50 @@
+--TEST--
+MongoDB\Driver\TopologyDescription::getServers()
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; ?>
+<?php skip_if_not_live(); ?>
+--FILE--
+<?php
+require_once __DIR__ . "/../utils/basic.inc";
+
+class TopologyDescriptionProvider implements MongoDB\Driver\Monitoring\SDAMSubscriber
+{
+    private $topologyDescription;
+
+    public function topologyChanged(MongoDB\Driver\Monitoring\TopologyChangedEvent $event)
+    {
+        $this->topologyDescription = $event->getNewDescription();
+    }
+
+    public function getTopologyDescription()
+    {
+        $manager = create_test_manager();
+        $manager->addSubscriber($this);
+        $manager->executeCommand(DATABASE_NAME, new MongoDB\Driver\Command(['ping' => 1]));
+        $manager->removeSubscriber($this);
+
+        return $this->topologyDescription;
+    }
+}
+
+function assertIsServerDescriptionsArray(array $sds) {
+    foreach ($sds as $sd) {
+        if (! $sd instanceof MongoDB\Driver\ServerDescription) {
+            return false;
+        }
+    }
+    return true;
+}
+
+$subscriber = new TopologyDescriptionProvider;
+$topologyDescription = $subscriber->getTopologyDescription();
+$serverDescriptions = $topologyDescription->getServers();
+
+var_dump(assertIsServerDescriptionsArray($serverDescriptions));
```

Comment:
```suggestion
var_dump(isArrayOfServerDescriptions($serverDescriptions));
```

Reply:
I think it also makes sense to assert that the array has at least one element in it (using [count](https://www.php.net/count)). Otherwise, it's possible that the array is empty and this still passes.

In practice, we wouldn't expect that to happen since the subscriber captures the final TD; however, the extra assertion will make the test more resilient if that were ever to change.